### DISCUSSION TITLE
Don't run action on both branch and pull request

### DISCUSSION THREAD

**Discussion by DanySK** [community/community#26389] [Actions]
<p>I’ve got a continuous integration workflow which is quite heavy, with multiplatform testing.<br>
I want it to run:</p>
<ul>
<li>On pushes, on any branch</li>
<li>On pull requests coming <em>from other repositories</em>
</li>
</ul>
<p>I currently enabled <code>push</code> and <code>pull_request</code>, but in case of pull requests from branches coming from the same repository, I get the build doubled. Since I already got a branch protection rule preventing merges of unsynced PRs, there is never the case in which I can have different results between the branch and pull request build.</p>
<p>How can I disable it in such case?</p>

**Comment by Simran-B** [comment, marked as answer]
<ul>
<li>You can rely on naming conventions to restrict <code>on: pull_request</code> to branches with a colon in them (typically just the case with PRs from forked repos)<br>
<a href="https://github.community/t/how-to-trigger-an-action-on-push-or-pull-request-but-not-both/16662/10" class="inline-onebox">How to trigger an action on push or pull request but not both?</a>
</li>
<li>You can skip jobs conditionally e.g. like so:<br>
<code>if: github.event_name == 'push' || (github.event_name == 'pull_request' &amp;&amp; github.repository != github.event.pull_request.head.repo.full_name)</code><br>
<a href="https://github.community/t/is-there-a-way-to-tell-if-a-pr-is-from-a-forked-repository/134186/7" class="inline-onebox">Is there a way to tell if a PR is from a forked repository?</a>
</li>
</ul>

**Comment by Simran-B** [comment]
<p>I suppose you could also restrict <code>push</code> to the default branch (<code>main</code>/<code>master</code>). <code>pull_request</code> will take care of pushes to other branches (type <code>synchronize</code>).</p>
